client/src/nonblocking.rs

```diff
@@ -160,14 +103,47 @@ impl<C: Deref<Target = impl Signer> + Clone> Program<C> {
         )
     }
 
-    pub async fn create_event_handler(&self) -> Result<EventHandler, ClientError> {
-        let ws_url = self.cfg.cluster.ws_url();
-        let client = PubsubClient::new(ws_url).await?;
+    /// Subscribe to program logs.
+    ///
+    /// Returns an [`EventUnsubscriber`] to unsubscribe and close connection gracefully.
+    pub async fn on<T: anchor_lang::Event + anchor_lang::AnchorDeserialize>(
+        &self,
+        f: impl Fn(&EventContext, T) + Send + 'static,
+    ) -> Result<EventUnsubscriber, ClientError> {
+        let client = Arc::new(PubsubClient::new(self.cfg.cluster.ws_url()).await?);
```

Comment:
The only problem is for the pubsubclient. Creating a new one and creating an Arc structure each time might be a bit resource intensive. Maybe we can just add a field that corresponds to an option in the program to initiate it at the first `on`

Reply:
Yeah, you are right on that and I think your solution is great.

Reply:
The only problem is that the program must be mutable in this case

Reply:
Can we not use interior mutability with `Arc<RwLock<T>>`?